src/util/time.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file time.js
+ */
+
+const findSegmentForTime = (time, timeline) => {
+  const matchingSegments = timeline.filter((x) => {
+    return x.start <= time && time <= x.end;
+  });
+
+  if (matchingSegments.length === 0) {
+    return null;
+  }
+
+  return matchingSegments[0].segment;
+};
+
+const buildMediaTimeline = (playlist) => {
+  const timeline = [];
+
+  if (!playlist.segments || playlist.segments.length === 0) {
+    return;
+  }
+
+  // Assumptions:
+  // - there will always be a segment.duration
+  // - we can start from zero
+  // - segments are in time order
+  // - segment.start and segment.end only come
```

Comment:
I know this may just be temporary (hence the "ignored for now"), but we are going to have to be careful here. If we haven't seen a segment, thus don't have start and end times, we are making an inaccurate guess, and we may need to surface that info somehow to allow a user to reject the returned time.

Reply:
Maybe I'm not entirely clear on what you mean here.

The segment.start and segment.end are appended to the playlist after running through sync-controller. I believe we have a later story to take the timestampOffset and discontinuities into account, which I think would take precedent over the segment.start and segment.end times, right?

Reply:
Ah, yeah. That's a good point too